diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0a2a552..2abae26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,10 +4,10 @@ project(qtengine_plugin LANGUAGES CXX)
 
 include(GNUInstallDirs)
 
-set(QT6_MIN_VERSION 6.9)
+set(QT6_MIN_VERSION 6.8)
 set(QT5_MIN_VERSION 5.15)
 set(KF5_MIN_VERSION "5.102.0")
-set(KF6_MIN_VERSION "6.20.0")
+set(KF6_MIN_VERSION "6.13.0")
 
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTOUIC ON)
diff --git a/src/qt6/common/config/configmanager.cpp b/src/qt6/common/config/configmanager.cpp
index d0b3639..958b444 100644
--- a/src/qt6/common/config/configmanager.cpp
+++ b/src/qt6/common/config/configmanager.cpp
@@ -9,7 +9,6 @@
 #include <qiodevice.h>
 #include <qjsondocument.h>
 #include <qjsonobject.h>
-#include <qjsonparseerror.h>
 #include <qjsonvalue.h>
 #include <qlist.h>
 #include <qlogging.h>
